pkg/resource/endpoint/hooks.go

```diff
@@ -10,9 +10,12 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/arn"
 	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/google/go-cmp/cmp/cmpopts"
 
 	svcsdk "github.com/aws/aws-sdk-go/service/eventbridge"
 
+	"github.com/google/go-cmp/cmp"
```

Comment:
This is up to you if you want to add this dependency. Please note that this library explicitly notes it should not be used for production code, though :)

> This package is intended to be a more powerful and safer alternative to reflect.DeepEqual for comparing whether two values are semantically equal. It is intended to only be used in tests, as performance is not a goal and it may panic if it cannot compare the values. Its propensity towards panicking means that its unsuitable for production environments where a spurious panic may be fatal.

https://pkg.go.dev/github.com/google/go-cmp/cmp#pkg-overview

Reply:
Changed to using `"golang.org/x/exp/slices"`, thx for pointing this out.